client/src/lib/schemaFileUtils.ts

```diff
@@ -58,66 +58,46 @@ export function replaceFileData(
   return innerReplaceFileDataWithURL(schema, schema.$defs, obj, replacer) as Record<string, unknown>;
 }
 
-export type FileFormat = 'audio' | 'image' | 'text' | 'document';
-
-function appendToFormats(
+export function requiresFileSupport(
   schema: JsonValueSchema | undefined,
-  defs: JsonSchemaDefinitions | undefined,
-  formats: Set<FileFormat>
-) {
+  defs: JsonSchemaDefinitions | undefined
+): boolean {
   if (schema === undefined || defs === undefined) {
-    return;
-  }
-
-  if (schema.followedRefName === 'Image') {
-    formats.add('image');
+    return false;
   }
 
-  if (schema.followedRefName === 'File') {
-    switch (schema.format) {
-      case 'audio':
-        formats.add('audio');
-        break;
-      case 'image':
-        formats.add('image');
-        break;
-      case 'pdf':
-      case 'text':
-      case 'document':
-        formats.add('document');
-        break;
-      default:
-        // TODO: how should we handle a file without a format?
-        break;
-    }
+  if (
```

Comment:
You should modify `FILE_REF_NAMES` to add `Document` and do `schema.followedRefName in FILE_REF_NAMES`

Also I don't think the Document ref name actually exists ?